src/frontend/src/optimizer/plan_node/stream_hash_agg.rs

```diff
@@ -124,11 +139,9 @@ impl PlanTreeNodeUnary for StreamHashAgg {
     }
 
     fn clone_with_input(&self, input: PlanRef) -> Self {
-        let logical = generic::Agg {
-            input,
-            ..self.logical.clone()
-        };
-        Self::new(logical, self.vnode_col_idx, self.row_count_idx)
+        let mut new = self.clone();
+        new.logical.input = input;
+        new
```

Comment:
we just want to maintain the eowc property here, but ti will skip all property's derivation including order and distribution. I think they are dangerous, maybe we should not do that.